src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -1033,7 +1021,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                     Return If(dl, MakeHashToken(precedingTrivia, fullWidth))
 
                 Case "&"c
-                    If CanGet(1) AndAlso BeginsBaseLiteral(Peek(1)) Then
+                    If Peep(1, ch) AndAlso BeginsBaseLiteral(ch) Then
```

Comment:
@TyOverby 
I'll re-examine this method.
Some of the case can alter the value of `ch` as all code path in that `case` exit from the function. It is the `case`'s that don't I have to be careful with as `ch` is used after the `Select` block.